src/libraries/Microsoft.Extensions.Hosting/src/HostApplicationBuilder.cs

```diff
@@ -182,45 +182,28 @@ private void Initialize(HostApplicationBuilderSettings settings, out HostBuilder
             logging = new LoggingBuilder(Services);
         }
 
-        /// <summary>
-        /// Provides information about the hosting environment an application is running in.
-        /// </summary>
+        IDictionary<object, object> IHostApplicationBuilder.Properties => _hostBuilderContext.Properties;
+
+        /// <inheritdoc />
         public IHostEnvironment Environment => _environment;
 
         /// <summary>
-        /// A collection of services for the application to compose. This is useful for adding user provided or framework provided services.
+        /// Gets the set of key/value configuration properties.
         /// </summary>
+        /// <remarks>
+        /// This can be mutated by adding more configuration sources, which will update its current view.
+        /// </remarks>
         public ConfigurationManager Configuration { get; }
```

Comment:
Should this now be typed as `IConfigurationManager`?

Reply:
No, that would be a breaking change.